### DISCUSSION TITLE
Expected server HTML to contain a matching <tr> in <table> / Hydration failed because the initial UI does not match what was rendered on the server

### DISCUSSION THREAD

**Discussion by romanusname** [vercel/next.js#46842] [Help] [answered]
### Summary

Next js 13.2.3
Next js cannot use the table element.

```
"use client";
export default function index() {
  return (
            <table>
                <tr>
                    <th>Company</th>
                    <th>Contact</th>
                    <th>Country</th>
                </tr>
                <tr>
                    <td>Alfreds Futterkiste</td>
                    <td>Maria Anders</td>
                    <td>Germany</td>
                </tr>
                <tr>
                    <td>Centro comercial Moctezuma</td>
                    <td>Francisco Chang</td>
                    <td>Mexico</td>
                </tr>
            </table>
  )
}
```
It works but I get an error.
```
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.
Warning: Expected server HTML to contain a matching <tr> in <table>.
```
Should I stop using this element or can I fix the problem?
https://stackblitz.com/edit/nextjs-jvjltx?file=app%2Fpage.js - example

### Example

https://stackblitz.com/edit/nextjs-jvjltx?file=app%2Fpage.js

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This is not a Next.js problem. It is something React is warning you about, a browser behavior.

You see the problem has to do with incorrect HTML. If you observe what the browser is trying to do, you'll see that it injects a `tbody` element!

- https://stackoverflow.com/questions/7490364/why-do-browsers-still-inject-tbody-in-html5

So you have to include a `tbody` as well! It must wrap your `tr` tags.

## Going deep

Now, if you want to see the `tbody` injected by the browser, you have to `disable JavaScript` so that React can't take over.

You'll see this:

![Screenshot 2023-03-06 at 20 15 11](https://user-images.githubusercontent.com/21013447/223208429-e20d877a-431a-445c-8540-8d44015dbd4d.png)

This part is fundamental for your learning career:

- This is not a Next.js error, 
- It is a React warning about browser behavior

> Notice that the error stack says: React, not Next.js

![Screenshot 2023-03-06 at 20 17 39](https://user-images.githubusercontent.com/21013447/223208629-002ce307-bd9e-4826-8490-9b930a72fd1d.png)

What browser behavior? AFAIK, browsers assume the presence of a `tbody` tag, https://alanhogan.com/code/implied-tbody, so when it is missing they fill it in.

When a request arrives to your server, React SSR responds with:

> Cut out scripts for readability, but notice there's no `tbody` tags

```html
<body><table><tr><th>Company</th><th>Contact</th><th>Country</th></tr><tr><td>Alfreds Futterkiste</td><td>Maria Anders</td><td>Germany</td></tr><tr><td>Centro comercial Moctezuma</td><td>Francisco Chang</td><td>Mexico</td></tr></table></body>
```

As soon as the browser parses that HTML it injects a tbody tag, and you get:

![Screenshot 2023-03-06 at 20 20 52](https://user-images.githubusercontent.com/21013447/223209320-34482194-e90f-4989-8110-e9f1a7259302.png)

If you were to inspect the source code of the page, you'd see what the server sent (no `tbody`!) It is only by disabling JavaScript that you can catch the browser misbehaving.

If JS had been enabled, wow React takes over and hydrates the page, but it finds a `tbody` inserted by the browser, where it expected a `tr` tag, and we got a problem, we cannot reliably hydrate, so a warning/error is printed out. 

Next.js catches that, and shows it in a modal for you.
